src/components/atoms/MonacoEditor/MonacoEditor.tsx

```diff
@@ -0,0 +1,98 @@
+import React, {lazy, useEffect, useState} from 'react';
+import type {EditorDidMount, monaco} from 'react-monaco-editor';
+
+import Colors from '@styles/Colors';
+
+const MonacoEditor = lazy(() => import('react-monaco-editor'));
+
+type TkMonacoEditorProps = {
+  language: string;
+  height?: string;
+  value: string;
+  onChange: (value: string) => void;
+};
+
+const lineHeight = 22;
+
+const options = {
+  contextmenu: true,
+  fontFamily: 'Roboto Mono, Monaco, monospace',
+  fontSize: 13,
+  lineHeight,
+  minimap: {
+    enabled: false,
+  },
+  scrollbar: {
+    horizontalSliderSize: 4,
+    verticalSliderSize: 18,
+  },
+  selectOnLineNumbers: true,
+  readOnly: false,
+  automaticLayout: true,
+  padding: {
+    top: 12,
+    bottom: 12,
+  },
+};
+
+const defaultEditorHeight = 300;
+const editorMaxHeight = window.screen.height * 0.5;
```

Comment:
Is that expected, or should it be based on viewport?
```suggestion
const editorMaxHeight = window.innerHeight * 0.5; // = 50vh
```
Besides, probably it should be part of `TkMonacoEditor` function.

Reply:
It needs to be part of the `TkMonacoEditor` to compute dynamically - otherwise, it will not react to the change of screen/viewport size.

Reply:
Got it, updating